project/accounts/views.py

```diff
@@ -97,104 +116,57 @@ def get_initial(self):
-                    "link": redirect_link,
-                }
-        else:
-            # invalid link
-            redirect_link = {"href": "/", "label": "Back to Main"}
-            template_var = {
-                "title": "Email Verification Error",
-                "content": "Email could not be verified",
-                "link": redirect_link,
-            }
-
-        return TemplateResponse(request, "general_message.html", template_var)
-
-
-class ProfileSetupView(LoginRequiredMixin, View):
-    """A view to make the user profile full_profile"""
-
-    login_url = "accounts_login"
-
-    def get(self, request):
-        profile = Profile.objects.get(user=request.user)
-        if profile.full_profile:
-            return HttpResponseRedirect("/")
-            # start temp rep rendering TODO: REMOVE THIS
-        else:
-            data = {
-                "username": request.user.username,
-                "email": request.user.email,
-            }
-            return TemplateResponse(request, "accounts/user-setup.html", data)
-
-
 class UserProfileView(LoginRequiredMixin, View):
     """A view that shows profile for authorized users"""
 
-    @method_decorator(full_profile)
     def get(self, request, username=None):
-        if not username:
-            return HttpResponseRedirect(f"/profile/{request.user}")
-        else:
-            is_owner = username == request.user.username
-            try:
-                user = get_user_model().objects.get(username=username)
-
-            except get_user_model().DoesNotExist:
-                return HttpResponseRedirect("/404")
-
-        form = ProfileEditForm(
-            initial={
-                "username": user.username,
-                "email": user.email,
-                "first_name": user.profile.first_name or None,
-                "last_name": user.profile.last_name or None,
-                "about_me": user.profile.about_me or None,
+        user_model = get_user_model()
+        user = get_object_or_404(user_model, username=username)
+
+        return TemplateResponse(
+            request,
+            "account.html",
+            {
+                "user": user,
             },
-            readonly=True,
         )
-        data = {
-            "username": user,
-            "profile_image_form": UpdateProfileImage,
-            "form": form if is_owner else None,
-            "readonly": True,
-        }
-        return TemplateResponse(request, "account.html", data)
+
+
+@login_required
+def expunge_user(request):
+    """
+    Delete User Information
+    """
+
+    user_model = get_user_model()
+    user = get_object_or_404(user_model, username=request.user.username)
+
+    profile = get_object_or_404(Profile, user=user)
+
+    # Expunge personally identifiable data in user
+    expunged_user_data = {
+        "is_active": False,
+        "email": "",
+        "first_name": "",
+        "last_name": "",
+        "username": f"expunged-{ user.id }",
```

Comment:
Changing the general approach to expunging user data is out-of-scope for this pull request since it was part of issue #227 and merged in pull request #1264.

We can open a new discussion to reappraise how we handle expunging user data when requesting their account to be deleted. Based on that discussion, we can open issues to improve the deletion/expungement feature.

Reply:
I wasn't aware of the user delete implementation. We can discuss the logic later. 👍

Reply:
All good. Your attention to detail is excellent. You've probably spotted a bug we can add to our backlog for a contributor to fix.